### PR TITLE
a note to migrate server

### PR DESCRIPTION
**Pull request by QwertyJack**
some useful note

### CODE REVIEW

**Comment by QwertyJack** [comment]
@huxuan @JeldorPKU

**Comment by JeldorPKU** [comment]
附议！

**Comment by QwertyJack** [comment]
@huxuan

**Comment by huxuan** [comment]
I will take a look soon.

File: migrat.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+### Migration
+by Jack
+2016年 12月 14日 星期三 11:14:48 CST
+
+# install php5.6
+```
+sudo apt install mysql-server apache2 apache2-utils phpmyadmin
+sudo add-apt-repository ppa:ondrej/php
+sudo apt update
+sudo apt-get install php7.0 php5.6 php5.6-mysql php-gettext php5.6-mbstring php-xdebug libapache2-mod-php5.6 libapache2-mod-php7.0
```
Comment (huxuan) [inline migrat.md]: Do we really need php7.0?

File: migrat.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+```
+mysql -uroot -p < foo.sql
+```
+
+# Apache minimal conf
+```
+<VirtualHost *:80> 
+   ServerAdmin webmaster@localhost 
+   DocumentRoot /<path>/<to>/CAPUBBS 
+   ServerName chexie.net
```
Comment (huxuan) [inline migrat.md]: How about supporting `*.chexie.net` as well?

File: vhost.conf.example [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	ServerAdmin webmaster@localhost
+	DocumentRoot <absolute-path-to-CAPUBBS>
+	ServerName <FQDN>
+	<Directory />
+		AddDefaultCharset UTF-8
+		<IfModule mod_authz_core.c>
+			# Apache 2.4
+			Require all granted
+		</IfModule>
+	    <IfModule !mod_authz_core.c>
```
Comment (huxuan) [inline vhost.conf.example]: Better indentation.

File: migrat.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+### Migration
+by Jack
+2016年 12月 14日 星期三 11:14:48 CST
```
Comment (huxuan) [inline migrat.md]: Suggest to use ISO 8601 format as here are the only Chinese characters.

[1] https://en.wikipedia.org/wiki/ISO_8601

File: migrat.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+sudo apt update
+sudo apt-get install php7.0 php5.6 php5.6-mysql php-gettext php5.6-mbstring php-xdebug libapache2-mod-php5.6 libapache2-mod-php7.0
+sudo apt install php5.6-curl php5.6-xml
+sudo a2dismod php7.0; sudo a2enmod php5.6; sudo service apache2 restart
+sudo update-alternatives --set php /usr/bin/php5.6
+```
+
+# database
+```
+mysql -uroot -p < foo.sql
```
Comment (huxuan) [inline migrat.md:24]: I suggest we create a standalone mysql user for CAPUBBS for better security.

File: vhost.conf.example [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+		</IfModule>
+	</Directory>
+	ErrorLog logs/chexie.net-error_log
+	CustomLog logs/chexie.net-access_log common
+	LoadModule bw_module    modules/mod_bw.so
+	BandWidthModule On
+	ForceBandWidthModule On
+	BandWidth all 100000
+</VirtualHost>
+
```
Comment (huxuan) [inline vhost.conf.example:20]: Redundant empty line.

**Review by huxuan — CHANGES_REQUESTED** [review]
All in all, it looks great. Just some minor issues commented, feel free to reach me in case of any problem.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+/bin/conf
```
Comment (huxuan) [inline .gitignore:1]: Why we need to ignore this file? as it's already exist.

File: migrat.md (hunk 4 above)
Comment (QwertyJack) [inline migrat.md]: this line is not necessary at all

File: migrat.md (hunk 1 above)
Comment (QwertyJack) [inline migrat.md]: nope

File: migrat.md (hunk 5 above)
Comment (QwertyJack) [reply migrat.md:24]: just for data sync, only excutes once

File: .gitignore (hunk 7 above)
Comment (QwertyJack) [reply .gitignore:1]: not secure; I would not have uploaded it; my fault.
How to purge it?

File: migrat.md (hunk 2 above)
Comment (QwertyJack) [reply migrat.md]: i do not like that
see ["is-a-wildcard-dns-record-bad-practice"](http://serverfault.com/questions/483576/is-a-wildcard-dns-record-bad-practice)

File: .gitignore (hunk 7 above)
Comment (JeldorPKU) [reply .gitignore:1]: @QwertyJack `git rm /bin/conf`
You have to manually delete it.

**Review by huxuan — APPROVED** [review]

**Pull request merged**